src/main/java/dao/BookRepositoryImpl.java

```diff
@@ -0,0 +1,43 @@
+package dao;
+
+import exception.DataProcessingException;
+import java.awt.print.Book;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+public class BookRepositoryImpl implements BookRepository {
+    private SessionFactory sessionFactory;
+
+    @Override
+    public Book save(Book book) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't save book to DB", e);

```

Comment:
Better use your custom EntityNotFoundException here

Reply:
I create it in the next task (we have it in description of our hw) so I left here DataProcessingException